internal/plugin/available_plugins.go

```diff
@@ -1,13 +1,81 @@
 package plugin
 
 import (
+	"github.com/cyberark/secretless-broker/pkg/secretless/log"
 	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector/http"
 	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector/tcp"
 )
 
+// CompatiblePluginAPIVersion indicates what matching API version an external plugin
+// must have so that we are capable of loading it.
+var CompatiblePluginAPIVersion = "0.1.0"
+
 // AvailablePlugins is an interface that provides a list of all the available
 // plugins for each type that the broker supports.
 type AvailablePlugins interface {
 	HTTPPlugins() map[string]http.Plugin
 	TCPPlugins() map[string]tcp.Plugin
 }
+
+// Plugins represent a holding object for a bundle of plugins of different types.
+type Plugins struct {
+	HTTPPluginMap map[string]http.Plugin
+	TCPPluginMap  map[string]tcp.Plugin
+}
+
+// HTTPPlugins returns only the HTTP plugins in the Plugins struct.
+func (plugins *Plugins) HTTPPlugins() map[string]http.Plugin {
+	return plugins.HTTPPluginMap
+}
+
+// TCPPlugins returns only the TCP plugins in the Plugins struct.
+func (plugins *Plugins) TCPPlugins() map[string]tcp.Plugin {
+	return plugins.TCPPluginMap
+}
+
+// AllAvailablePlugins returns the full list of internal and external plugins
+// available to the broker.
+// TODO: Test this
+func AllAvailablePlugins(pluginDir string,
```

Comment:
nit: indent the args the same ? 
as you did with:
```
func ExternalPlugins(
	pluginDir string,
```

maybe we can agree on a style and be consistent with it